### DISCUSSION TITLE
call curl_easy_perform()  error ----- malloc(): invalid size(unsort)

### DISCUSSION THREAD

**Discussion by Docin123** [curl/curl#8031] [Q&A] [answered]
I do cross-compile libcurl ,the host is ubuntu:20.04, target is raspbian10.
cross-compile tool is  gcc-10-arm-linux-gnueabihf.
when I finish cross-compile of libcurl, I cross-compile my main.c with libcurl.  compile success
but when it run on raspberry ,  it run error and give message " malloc(): invalid size(unsort)   "

**Comment by jzakrzewski** [comment, marked as answer]
I'd say start by running it with Valgrind. It's typically a sign of some out-of-bounds write, which Valgrind could help to pinpoint.
